# Generated by Django 5.2 on 2025-05-10 14:54

import django.utils.timezone
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('tracker', '0024_userprofile_mfa_enabled_userprofile_phone_number'),
    ]

    operations = [
        migrations.CreateModel(
            name='CriticalErrorLog',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('timestamp', models.DateTimeField(db_index=True, default=django.utils.timezone.now, verbose_name='Timestamp')),
                ('level_name', models.CharField(help_text='e.g., CRITICAL, ERROR', max_length=50, verbose_name='Log Level Name')),
                ('message', models.TextField(help_text='The main error message.', verbose_name='Error Message')),
                ('pathname', models.CharField(blank=True, help_text='Full pathname of the source file where the logging call was issued.', max_length=1024, null=True, verbose_name='Pathname')),
                ('funcName', models.CharField(blank=True, help_text='Name of function containing the logging call.', max_length=255, null=True, verbose_name='Function Name')),
                ('lineno', models.PositiveIntegerField(blank=True, help_text='Source line number where the logging call was issued.', null=True, verbose_name='Line Number')),
                ('traceback_info', models.TextField(blank=True, help_text='Full traceback, if available.', null=True, verbose_name='Traceback Information')),
                ('is_reviewed', models.BooleanField(default=False, verbose_name='Reviewed')),
            ],
            options={
                'verbose_name': 'Critical System Error Log',
                'verbose_name_plural': 'Critical System Error Logs',
                'ordering': ['-timestamp'],
            },
        ),
    ]
